docs/explanations/curator_data_model.md

```diff
@@ -0,0 +1,389 @@
+- `list`: Make sure you are using one of the list-types in the `columnType` column.
+- `regex`: `regex <module> <pattern>`, move the `<pattern>` to the `Pattern` column.
+- `inRange`: `inRange <minimum> <maximum>`, move the `<minimum>` and/or the `<maximum>` to the `Minimum` and `Maximum` columns respectively.
+- `date`: Use the `Format` column with value `date`
+- `url`: Use the `Format` column with value `uri`
+
+## Conditional dependencies
+
+The `DependsOn` and `Valid Values` columns can be used together to flexibly define conditional logic for determining the relevant attributes for a data type.
+
+In this example we have the `Patient` data type. In this case the `Patient` can dbe diagnosed as healthy or with cancer. For Patients with cancer we also want to collect info about their cancer type, and any cancers in their family history.
+
+Data Model:
+
+| Attribute      | DependsOn                      | Valid Values        | Required | columnType  |
+|---|---|---|---|---|
+| Patient        | "Diagnosis"                    |                     |          |             |
+| Diagnosis      |                                | "Healthy, Cancer"   | True     | string      |
+| Cancer         | "Cancer Type, Family History"  |                     |          |             |
+| Cancer Type    |                                | "Brain, Lung, Skin" | True     | string      |
+| Family History |                                | "Brain, Lung, Skin" | True     | string_list |
+
+To demonstrate this, see the above example with the `Patient` and `Cancer` data types:
+
+- `Diagnosis` is an attribute of `Patient`.
+- `Diagnosis` has `Valid Values` of `Healthy` and `Cancer`.
+- `Cancer` is also a data type.
+- `Cancer Type` and `Family History` are attributes of `Cancer` and are both required.
+
+As a result of the above data model, in the JSON Schema:
+
+- `Cancer Type` and `Family History` become properties of `Patient`.
+- For a given `Patient`, if `Diagnosis` == `Cancer` then `Cancer Type` and `Family History` become required for that `Patient`.
+- The conditional logic is contained in the `allOf` array.
+
+```JSON
+{
+  "description": "TBD",
+  "properties": {
+    "Diagnosis": {
+      "description": "TBD",
+      "enum": ["Cancer", "Healthy"],
+      "title": "Diagnosis",
+      "type": "string"
+    },
+    "Cancer Type": {
+      "description": "TBD",
+      "enum": ["Breast","Lung","Skin"],
+      "title": "Cancer Type",
+      "type": "string"
+    },
+    "Family History": {
+      "description": "TBD",
+      "title": "Family History",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": ["Breast","Lung","Skin"],
+      }
+    }
+  },
+  "required": ["Diagnosis"],
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "Diagnosis": {
+            "enum": [
+              "Cancer"
+            ]
+          }
+        }
+      },
+      "then": {
+        "required": ["Cancer Type", "FamilyHistory"]
```

Comment:
Here it should be "FamilyHistory" or "Family History"? It might be worth regenerating all the json schema examples